test/e2e/e2e_suite_test.go

```diff
@@ -103,26 +101,30 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 	if e2eConfig.GetVariable("DEPLOY_CERT_MANAGER") != "false" {
 		// Install cert-manager
 		By("Installing cert-manager")
-		cmd := exec.Command("cmctl", "check", "api")
-		_, err := cmd.CombinedOutput()
+		err := checkCertManagerAPI(clusterProxy)
 		if err != nil {
-			cmd = exec.Command("cmctl", "x", "install")
-			output, err := cmd.CombinedOutput()
-			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
-			By("Checking that the cert-manager API is available")
-			cmd = exec.Command("cmctl", "check", "api")
-			output, err = cmd.CombinedOutput()
-			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
+			cmVersion := e2eConfig.GetVariable("CERT_MANAGER_VERSION")
+			err = installCertManager(ctx, clusterProxy, cmVersion)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for cert-manager webhook")
+			Eventually(func() bool {
+				err := checkCertManagerWebhook(ctx, clusterProxy)
+				return err == nil
+			}, e2eConfig.GetIntervals("default", "wait-available")...).Should(BeTrue())
```

Comment:
I may be missing something, but is it not possible to just return the error directly and way for `Succeed` instead?
```suggestion
			Eventually(func() err {
				return checkCertManagerWebhook(ctx, clusterProxy)
			}, e2eConfig.GetIntervals("default", "wait-available")...).Should(Succeed())
```

Reply:
Thanks a lot! That's the way to go :sweat_smile: I'm still pretty not familiar yet with the CAPI test framework. Initially used `BeNil()` as the function returns an error, and the wait just never worked :sweat_smile: